src/FixedDataTable.js

```diff
@@ -508,43 +508,77 @@ class FixedDataTable extends React.Component {
     this._didScrollStop = debounceCore(this._didScrollStopSync, 200, this);
     this._onKeyDown = this._onKeyDown.bind(this);
 
-    this._wheelHandler = new ReactWheelHandler(
-      this._onScroll,
-      this._shouldHandleWheelX,
-      this._shouldHandleWheelY,
-      this.props.isRTL,
-      this.props.stopScrollDefaultHandling,
-      this.props.stopScrollPropagation
-    );
-
-    this._touchHandler = new ReactTouchHandler(
-      this._onScroll,
-      this._shouldHandleTouchX,
-      this._shouldHandleTouchY,
-      this.props.stopScrollDefaultHandling,
-      this.props.stopScrollPropagation
-    );
+    this._setupHandlers();
   }
 
   componentWillUnmount() {
+    this._cleanupHandlers();
+
+    // Cancel any pending debounced scroll handling and handle immediately.
+    this._didScrollStop.reset();
+    this._didScrollStopSync();
+  }
+
+  _setupHandlers() {
+    if (!this._wheelHandler) {
+      this._wheelHandler = new ReactWheelHandler(
+        this._onScroll,
+        this._shouldHandleWheelX,
+        this._shouldHandleWheelY,
+        this.props.isRTL,
+        this.props.stopScrollDefaultHandling,
+        this.props.stopScrollPropagation
+      );
+    }
+
+    if (!this._touchHandler) {
+      this._touchHandler = new ReactTouchHandler(
+        this._onScroll,
+        this._shouldHandleTouchX,
+        this._shouldHandleTouchY,
+        this.props.stopScrollDefaultHandling,
+        this.props.stopScrollPropagation
+      );
+    }
+
     // TODO (pradeep): Remove these and pass to our table component directly after
     // React provides an API where event handlers can be specified to be non-passive (facebook/react#6436)
-    this._divRef &&
-      this._divRef.removeEventListener('wheel', this._wheelHandler.onWheel, {
+    if (this._divRef) {
+      this._divRef.addEventListener('wheel', this._wheelHandler.onWheel, {
```

Comment:
We don't need a guard check here cause the event handler is a static function reference, so multiple `addEventListener` calls don't [add duplicate entries](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#memory_issues).